app/messaging/process-backoffice.js

```diff
@@ -0,0 +1,31 @@
+const util = require('util')
+const { getAll, getApplicationCount } = require('../repositories/application-repository')
+const { backOfficeResponseMsgType, backOfficeResponseQueue } = require('../config')
+const sendMessage = require('../messaging/send-message')
+const states = require('./states')
+
+const processBackOfficeRequest = async (msg) => {
+  const responseMessage = msg.body
+  const { sessionId } = msg
+  try {
+    console.log('BackOffice Request received:', util.inspect(msg.body, false, null, true))
+    // Get ID
+    const result = await getAll(msg.body.limit ?? 10, msg.body.offset ?? 0, msg.body.search.text)
+    const total = await getApplicationCount(msg.body.search.text)
+    if (result.length <= 0) {
+      await sendMessage({ applicationState: states.notFound, applications: [], total: 0 }, backOfficeResponseMsgType, backOfficeResponseQueue, { sessionId })
```

Comment:
I don't think `applicationState` adds any value here? I think this `if` block could just be, using the `total` from the query. I presume `result` is an empty array when there are no applications?

```js
await sendMessage({ applications: result, total }, backOfficeResponseMsgType, backOfficeResponseQueue, { sessionId })
```

Reply:
We will change this in next PR.

Reply:
Ok, can you add an issue to the repo so we can track it gets done please?

Reply:
Sure.

Reply:
#26